[glfs-fops.c]: minor improvements

pbhatia22 opened the pull request:
Converted calloc to malloc as already initialized
all fields after allocation and make changes to ensure
initialization happens in the order variables are in the struct

Updates: #1000
Change-Id: I59784d4e2d2b2903e93724350524824f6447793b
Signed-off-by: Preet Bhatia <pbhatia@redhat.com>

pbhatia22 committed:
[glfs-fops.c]: minor improvements

Converted calloc to malloc as already initialized
all fields after allocation and make changes to ensure
initialization happens in the order variables are in the struct

Change-Id: I59784d4e2d2b2903e93724350524824f6447793b
Updates: #1000
Signed-off-by: Preet Bhatia <pbhatia@redhat.com>

pbhatia22 commented:
/run regression

xhernandez commented on api/src/glfs-fops.c:
```diff
@@ -5681,14 +5681,13 @@ glfs_recall_lease_upcall(struct glfs *fs, struct glfs_upcall *up_arg,
     }
 
-    up_lease_arg = GF_CALLOC(1, sizeof(struct glfs_upcall_lease),
+    up_lease_arg = GF_MALLOC(sizeof(struct glfs_upcall_lease),
```
Please, also add a check to verify that allocation didn't fail.

pbhatia22 replied:
Added

pbhatia22 committed:
[glfs-fops.c]: minor improvements
Converted calloc to malloc as already initialized
all fields after allocation and make changes to ensure
initialization happens in the order variables are in the struct

Updates: #1000
Change-Id: I59784d4e2d2b2903e93724350524824f6447793b
Signed-off-by: Preet Bhatia <pbhatia@redhat.com>

pbhatia22 commented:
/run regression

xhernandez commented on api/src/glfs-fops.c:
```diff
@@ -5681,14 +5681,17 @@ glfs_recall_lease_upcall(struct glfs *fs, struct glfs_upcall *up_arg,
     up_lease_arg->object = object;
+    up_lease_arg->lease_type = recall_lease->lease_type;
 
     GF_VALIDATE_OR_GOTO("glfs_recall_lease", up_lease_arg, out);
```
This is useless here. It should be removed.

pbhatia22 replied:
Are you talking about line number 5693 ?

pbhatia22 committed:
Merge branch 'gluster:devel' into glfs_improvement

pbhatia22 committed:
[glfs-fops.c]: minor improvements

Converted calloc to malloc as already initialized
all fields after allocation and make changes to ensure
initialization happens in the order variables are in the struct

Change-Id: I59784d4e2d2b2903e93724350524824f6447793b
Updates: gluster#1000
Signed-off-by: Preet Bhatia <pbhatia@redhat.com>

xhernandez commented:
/run regression
